### ISSUE TITLE
Segfault caused by increase timeout sanity check

### ISSUE THREAD

**Issue by pparadzik** [warmcat/libwebsockets#1081]
I get this
```
==10759== Invalid read of size 8
==10759==    at 0x4E5C328: lws_service_fd_tsi (service.c:1050)
==10759==    by 0x4E5D764: lws_service_fd (service.c:1709)
```
and it seems that it's caused by this commit
```
commit f1d70692cd46f8e2b91c6591d0ea439e2a2a65ab
Author: Andy Green <andy@warmcat.com>
Date:   Tue Oct 31 07:02:51 2017 +0800

    ah: increase timeout sanity check to ah_idle + 60

diff --git a/lib/service.c b/lib/service.c
index 1de44ad7..b34c0505 100644
--- a/lib/service.c
+++ b/lib/service.c
@@ -1042,7 +1042,8 @@ lws_service_fd_tsi(struct lws_context *context, struct lws_pollfd *pollfd, int t
                        const unsigned char *c;
 
                        if (!ah->in_use || !ah->wsi || !ah->assigned ||
-                           now - ah->assigned < 60) {
+                           now - ah->assigned <
+                           wsi->vhost->timeout_secs_ah_idle + 60) {
                                ah = ah->next;
                                continue;
                        }
```
If I comment out `wsi->vhost->timeout_secs_ah_idle`, it works ok.

**Comment by lws-team** [comment]
Sorry I fixed it thismorning, but evidently didn't push master since then.

I updated master just now.

BTW are you interested in ACME / Letsencrypt automated cert provisioning / renewal via tls-sni-01 / -02?  Because I nearly have this working with openssl backend, but I am not sure mbedtls backend can do it.  And soon I am going to have to do other things.

If it's interesting there are just two functions needed that exist in `lib/tls/openssl/server.c` that need replicating in the mbedtls version, `lws_tls_acme_sni_cert_create()` and `lws_tls_acme_sni_csr_create()`.  The main problem is mbedtls does not seem to understand the OID for the SAN (Subject Alternate Names) needed for the SNI challenge.
